projects/igniteui-angular/src/lib/dialog/dialog.component.ts

```diff
@@ -64,50 +64,188 @@ export class IgxDialogComponent implements IToggleView, OnInit, OnDestroy {
     private static NEXT_ID = 1;
     private static readonly DIALOG_CLASS = 'igx-dialog';
 
+    /**
+    * An @Input property that sets the value of the `id` attribute.
+    *```html
+    *<igx-dialog [id]="'igx-dialog-56'" #alert title="Notification" leftButtonLabel="OK" (onLeftButtonSelect)="alert.close()"></igx-dialog>
+    *```
+    */
     @HostBinding('attr.id')
     @Input()
     public id = `igx-dialog-${DIALOG_ID++}`;
 
+    /**
+    * An @Input property controlling the `title` of the dialog.
+    *```html
+    *<igx-dialog title="Notification" #alert leftButtonLabel="OK" (onLeftButtonSelect)="alert.close()"></igx-dialog>
+    *```
+    */
     @Input()
     public title = '';
+
+    /**
+     *  An @Input property controlling the `message` of the dialog.
+     *```html
+     *<igx-dialog message="Your email was sent!" #alert leftButtonLabel="OK" (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public message = '';
 
+    /**
+     * An @Input property to set the `label` of the left button of the dialog.
+     *```html
+     *<igx-dialog leftButtonLabel="OKAY" #alert title="Notification"  (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public leftButtonLabel = '';
+
+    /**
+     * An @Input property to set the left button `type`. The types are `flat`, `raised` and `fab`.
+     * The `flat` type button is a rectangle and doesn't have a shadow. <br>
+     * The `raised` type button is also a rectangle but has a shadow. <br>
+     * The `fab` type button is a circle with a shadow. <br>
+     * The default value is `flat`.
+     *```html
+     *<igx-dialog leftButtonType="raised" leftButtonLabel="OKAY" #alert (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public leftButtonType = 'flat';
+    /**
+     * An @Input property to set the left button color.
```

Comment:
@alexalex972 It's good to clarify whether the user can use hexadecmal values or RGB values. For example: "The property accepts all valid CSS `color` property values.